SalesforceCommerceCloud/pwa-kit
# [Temp Ext] Refactor overrides system to use new `apply(resolver)` solution

# Description
In performance benchmarking, it was discovered that the previous solution, in a worst-case scenario of someone copying ALL of the files from their project into the overrides directory, took about **2m 22s** for a webpack initial build. This is degraded from ~14s to run the current version of the retail-react-app. This level of degradation is considered a significant blocker for customer adoption, as many will likely copy in a huge number of files and hit this performance degradation.

This PR rewrites the webpack logic to tap into the `resolve` hook in a new webpack plugin.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Created a new Webpack plugin to implement the overrides system using the `

# How to Test-Drive This PR

- Run `npm start` from the `spike-extendend-retail-app` directory to verify the overrides system is working as expected

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
